### PR TITLE
add request parameter restriction to documentation

### PR DESCRIPTION
**Pull request by dzanot**
## What does this change do?
- Clarifies the existing skeuomorph restrictions on single argument RPC calls. 
- Updates the generated code examples to use `List` instead of `Seq`

## Checklist

- [ ] Reviewed the diff to look for typos, println and format errors.
- [ ] Updated the docs accordingly.

### CODE REVIEW

File: microsite/src/main/docs/reference/source-generation.md [hunk 1]
```diff
@@ -161,15 +162,26 @@ record SearchRequest {
...
+This schema can be evolved to add optional fields (e.g. ordering, filters, ...) to the request.  All the user has to do is just change the _single record_.  
 
 This API design, on the other hand, can't be evolved because changing the `SearchResponse` argument from a `string` to any other datatype would introduce backward incompatibility.
 
 ```avroidl
 SearchResponse search(string query);
 ```
 
-For this reason, we **generally encourage that all requests and responses must be records**.
+Similarly, multiple arguments don't fully restrict api evolutions but can become inconsistent. Consider,
```
Comment (L-Lavigne) [inline microsite/src/main/docs/reference/source-generation.md:173]: Let's capitalize API here.

File: microsite/src/main/docs/reference/source-generation.md [hunk 2]
```diff
@@ -161,15 +162,26 @@ record SearchRequest {
...
+
+```avroidl
+record Filter {
+  array<string> exclude;
+}
+SearchResponse search(SearchRequest query, Filter filter)
+```
+
+If we wanted to add a way to order the results and add it to our `SearchRequest`, it starts to make less sense to have `filter` be it's own argument.
```
Comment (L-Lavigne) [inline microsite/src/main/docs/reference/source-generation.md:182]: it's -> its

**Review by L-Lavigne — COMMENTED** [review]
Could we add a PR description and tags? Thanks!

File: microsite/src/main/docs/reference/source-generation.md (hunk 1 above)
Comment (dzanot) [reply microsite/src/main/docs/reference/source-generation.md]: Done!

File: microsite/src/main/docs/reference/source-generation.md (hunk 2 above)
Comment (dzanot) [reply microsite/src/main/docs/reference/source-generation.md]: Thanks, done!

File: microsite/src/main/docs/reference/source-generation.md [hunk 3]
```diff
@@ -120,9 +120,10 @@ sbt module containing the IDL definitions (`foo-domain`):
 ## Implementation Notes: An Intentional Incompatibility with the Avro Standard
 
 In order to make it so that it's easier for users to evolve their schemas over time, 
-`sbt-mu-srcgen` intentionally deviates from the Avro standard in one key way: it does 
-not permit primitive types (e.g. `string sendUser(UserWithCountry user)`) to be present 
-in the Avro schema.  If you attempt to write an Avro schema using primitive types instead
+`sbt-mu-srcgen` intentionally deviates from the Avro standard in one key way: 
+it restricts RPC return types to record types (`string sendUser(UserWithCountry user)` is not permitted)
```
Comment (dmarticus) [inline microsite/src/main/docs/reference/source-generation.md:124]: this is a clearer change, I like it!

File: microsite/src/main/docs/reference/source-generation.md [hunk 4]
```diff
@@ -161,15 +162,26 @@ record SearchRequest {
...
+```avroidl
+record Filter {
+  array<string> exclude;
+}
+SearchResponse search(SearchRequest query, Filter filter)
+```
+
+If we wanted to add a way to order the results and add it to our `SearchRequest`, it starts to make less sense to have `filter` be its own argument. 
+
+For this reason, we **enforce that all requests and responses must be records**.
```
Comment (dmarticus) [inline microsite/src/main/docs/reference/source-generation.md:184]: hell yeah lock it down

File: microsite/src/main/docs/reference/source-generation.md [hunk 5]
```diff
@@ -161,15 +162,26 @@ record SearchRequest {
...
+Similarly, multiple arguments don't fully restrict API evolutions but can become inconsistent. Consider,
+
+```avroidl
+record Filter {
+  array<string> exclude;
+}
+SearchResponse search(SearchRequest query, Filter filter)
+```
+
+If we wanted to add a way to order the results and add it to our `SearchRequest`, it starts to make less sense to have `filter` be its own argument.
```
Comment (dmarticus) [inline microsite/src/main/docs/reference/source-generation.md:182]:
```suggestion
If we wanted to add a way to order the results and add it to our `SearchRequest`, it doesn't make sense to have `filter` be its own argument.
```

**Review by dmarticus — APPROVED** [review]
one tiny wording nit but generally LGTM!

**Review by L-Lavigne — DISMISSED** [review]

File: microsite/src/main/docs/reference/source-generation.md [hunk 6]
```diff
@@ -146,8 +147,8 @@ message:
 ```
 [error] (protocol / muSrcGen) One or more IDL files are invalid. Error details:
 [error]  /path/to/the/invalid/file.avdl has the following errors:
-RPC method request parameter 'user' has non-record request type 'STRING', 
-RPC method response parameter has non-record response type 'STRING'
+Encountered an unsupported request type: Skeuomorph only supports Record types for Avro requests. Encountered request schema with type STRING
+Encountered an unsupported request type: Skeuomorph only supports Record types for Avro responses. Encountered response schema with type STRING
```
Comment (L-Lavigne) [inline microsite/src/main/docs/reference/source-generation.md:151]:
```suggestion
Encountered an unsupported response type: Skeuomorph only supports Record types for Avro responses. Encountered response schema with type STRING
```

**Review by juanpedromoreno — APPROVED** [review]

**Pull request merged**